### PR TITLE
430 use `expect_no_error` in tests instead of `expect_error(, NA)`

### PR DESCRIPTION
**Pull request by m7pr**
Part of https://github.com/insightsengineering/coredev-tasks/issues/430

### CODE REVIEW

File: tests/testthat/test-register_logger.R [hunk 1]
```diff
@@ -39,24 +39,23 @@ testthat::test_that("An additional logging namespace is created after register_l
...
 
 testthat::test_that("register_logger does not throw if passed a correct logger level", {
   withr::with_options(
     new = list(teal.log_layout = "{msg}"),
-    code = testthat::expect_error(register_logger(namespace = "test", level = logger::INFO), NA)
+    code = testthat::expect_no_error(register_logger(namespace = "test", level = logger::INFO))
   )
 })
 
 testthat::test_that("register_logger does not throw if passed NULL to either log_layout or log_level
```
Comment (averissimo) [inline tests/testthat/test-register_logger.R:53]: Linter errors, the code below should solve both

```r
testthat::test_that(
  "register_logger does not throw if passed NULL to either log_layout or log_level and valid options are set",
  {
    withr::with_options(
      new = list(teal.log_layout = "{msg}", teal.log_level = logger::INFO),
      code = testthat::expect_no_error(
        register_logger(namespace = "test", level = NULL, layout = NULL)
      )
    )
  }
)

testthat::test_that(
  paste(
    "register_logger",
    "does not throw if passed NULL to arguments, options are not set and the system variables are set to valid value"
  ), {
    withr::local_envvar(.new = list(TEAL.LOG_LAYOUT = "{msg}", TEAL.LOG_LEVEL = "INFO"))
    withr::local_options(.new = list(teal.log_layout = NULL, teal.log_level = NULL))
    testthat::expect_no_error(register_logger(namespace = "test", layout = NULL, level = NULL))
  }
)
```

**Review by averissimo — APPROVED** [review]
LGTM.

There are linter errors that are blocking the merge.

I gave a suggestion, feel free to merge after that is resolved

**Review by averissimo — CHANGES_REQUESTED** [review]
Please bump {testthat} version to 3.1.5

File: tests/testthat/test-register_logger.R (hunk 1 above)
Comment (m7pr) [reply tests/testthat/test-register_logger.R]: fixed in here -thanks cf9b27951cc3feba03be863492ad845511134354

**Comment by m7pr** [comment]
updated testthat version! thanks @averissimo

**Review by averissimo — APPROVED** [review]
:100:

**Pull request merged**